compiler/noirc_frontend/src/parser/parser/doc_comments.rs

```diff
@@ -1,36 +1,66 @@
-use chumsky::Parser;
-
-use crate::{
-    parser::NoirParser,
-    token::{DocStyle, Token, TokenKind},
-};
-
-use super::primitives::token_kind;
-
-fn outer_doc_comment() -> impl NoirParser<String> {
-    token_kind(TokenKind::OuterDocComment).map(|token| match token {
-        Token::LineComment(comment, Some(DocStyle::Outer)) => comment,
-        Token::BlockComment(comment, Some(DocStyle::Outer)) => comment,
-        _ => unreachable!(
-            "Parser should have already errored due to token not being an outer doc comment"
-        ),
-    })
-}
+use crate::token::{DocStyle, Token, TokenKind};
 
-pub(super) fn outer_doc_comments() -> impl NoirParser<Vec<String>> {
-    outer_doc_comment().repeated()
-}
+use super::Parser;
+
+impl<'a> Parser<'a> {
+    /// InnerDocComments = inner_doc_comment*
+    pub(super) fn parse_inner_doc_comments(&mut self) -> Vec<String> {
```

Comment:
In the end there weren't that many uses of `many0` (I think) but I'll still try to crate a helper function for this. I wanted to avoid as many `loop` and `while` in the parser to avoid accidentally looping forever.

Reply:
After looking at `parse_many` it looks like you could add a helper for that with no separator